apps/jobs/tasks.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import requests
+import shutil
+import tempfile
+import codecs
+
+from challenges.models import (
+    Challenge,
+    ChallengePhase
+)
+from django.contrib.auth.models import User
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.http import HttpRequest
+from evalai.celery import app
+from participants.models import ParticipantTeam
+from participants.utils import (
+    get_participant_team_id_of_user_for_a_challenge
+)
+from .models import Submission
+from .serializers import SubmissionSerializer
+from .utils import get_file_from_url
+from .sender import publish_submission_message
+
+logger = logging.getLogger(__name__)
+
+
+@app.task
+def download_file_and_publish_submission_message(
+    request_data,
+    user_pk,
+    request_method,
+    challenge_id,
+    challenge_phase_id
+):
+    """
+    Download submission file from url and send it for the evaluation
+    """
+    user = User.objects.get(pk=user_pk)
+    challenge = Challenge.objects.get(pk=challenge_id)
```

Comment:
I think we can get the challenge id using the `challenge_phase`. So, there won't be any need for this query.

Reply:
But we need this query in order to get the `challenge_phase` object for that particular challenge.